monitoring/queuemon_test.go

```diff
@@ -81,21 +85,22 @@ func (tq *TestMetricsQueue) Metrics() (queue.Metrics, error) {
 	return tq.metricState, nil
 }
 
-// ============ mocked output reporter
+func getRandPort() int {
+	return rand.Intn(65535-49152) + 49152 //nolint:gosec //This is a test, strong crypto not needed
```

Comment:
Any reason you can't just bind to port 0? That is exactly what httptest does: https://cs.opensource.google/go/go/+/refs/tags/go1.18.1:src/net/http/httptest/server.go;l=68;drc=refs%2Ftags%2Fgo1.18.1

If you want the allocated port value you'd have to get it out of the net.Listener (again httptest does this in a nicer way): https://stackoverflow.com/a/43425461

This doesn't guarantee you get a free port, just a random port.

Reply:
Ah, didn't even think of that.